templates/internship_detail.html

```diff
@@ -34,44 +34,107 @@
+                        </a>
+                        <div class="collapse" id="choice{{ forloop.counter0 }}">
+                            <ul style="list-style:none; padding-left: 0;">
+                                {% for student in students %}
+                                    {% if student.choice == forloop.parentloop.counter0|add:1 %}
+                                        <li>
+                                            <span style="margin-right: 5px;" class="glyphicon glyphicon-user text-primary"></span>
+                                            {{ student.student.person.first_name.0 }}. {{ student.student.person.last_name }}
+                                        </li>
+                                    {% endif %}
+                                {% endfor %}
+                            </ul>
+                        </div>
+                    <br/>
+                    <br>
+                {% endfor %}
+                </div>
+            </div>
+            </div>
+        {% endif %}
+            {% if non_mandatory_internships_choices %}
+            {% for i in number_choices_non_mandatory %}
+                <div class="col-lg-2">
+                <div class="panel panel-info">
+                <div class="panel-heading">
+                <h4>{% trans 'Non mandatory internship' %} - {{ forloop.counter }}</h4>
+                </div>
+                <div class="panel-body">
+                {% for j in i %}
+                    {% if forloop.counter0 == 0 %}
+                        <label> {% trans 'First choice' %} </label> : {{j}}<br/>
+                        {% elif forloop.counter0 == 1 %}
+                        <label> {% trans 'Second choice' %} </label> : {{j}}<br/>
+                        {% elif forloop.counter0 == 2 %}
+                        <label> {% trans 'Third choice' %} </label> : {{j}}<br/>
+                        {% elif forloop.counter0 == 3 %}
+                        <label> {% trans 'Fourth choice' %} </label> : {{j}}<br/>
+                    {% endif %}
+                    {% trans 'Students list' %} :
+                    <a role="button" data-toggle="collapse" data-target="#nm-choice-{{forloop.parentloop.counter0}}-{{ forloop.counter0 }}"
+                       aria-expanded="false" aria-controls="nmchoice-{{forloop.parentloop.counter0}}{{ forloop.counter0 }}"
+                       title="{% trans 'Display students' %}">
+                        <span class="glyphicon glyphicon-list" aria-hidden="true"></span>
+                    </a>
+                    <div class="collapse" id="nm-choice-{{forloop.parentloop.counter0}}-{{ forloop.counter0 }}">
+                        <ul style="list-style:none; padding-left: 0;">
+                            {% for student in nm_students %}
+                                    {% if student.internship.name|make_list|last|add:0 == forloop.parentloop.parentloop.counter|add:0 and student.choice == forloop.parentloop.counter %}
```

Comment:
student.internship.name|slice:"-1:" == forloop.parentloop.parentloop.counter and ...

Voir carrément faire un with forloop.parentloop as parentLoop ou un truc du genre pour pas répeter les .parentloop